src/Uri.php

```diff
@@ -143,7 +143,7 @@ public function withScheme($scheme): self
     public function withUserInfo($user, $password = null): self
     {
         $info = $user;
-        if ('' != $password) {
+        if ('' !== $password) {
```

Comment:
Password can be null

Reply:
Good catch.

Something that I didn’t even think about: is empty string (`''`) a valid HTTP password? I.e. is `https://zegnat:@example.com/` different from `https://zegnat@example.com/`? If an empty string is valid, this if-clause should specifically check `null !== $password`.

Reply:
I think it is valid, but I'm not sure it's any different from an no password at all. One could check the RFCs, but my gut tells that implementations might not always follow the RFC here.

Reply:
A quick look at [RFC 3986](https://tools.ietf.org/html/rfc3986), as cited by PSR-7, tells me that an empty string indicates no password. So really, `''` equals `null` as far as URIs are concerned. This also matches with [the WHATWG URL Standard](https://url.spec.whatwg.org/) (that tries to bring several other standards together), where [password is an empty string by default](https://url.spec.whatwg.org/#concept-url-password).

Part of me wants to argue that the method signature is wrong, and should have been `withUserInfo($user, $password = '')`.